test/unit/mods.example.ts

```diff
@@ -529,12 +529,100 @@ export const ValidTypeCombinations = new Map<string, ExampleMod>(
   }),
 );
 
+export const JsonMod = new Map<string, ExampleMod>(
+  Object.entries({
+    jsonWithValidFileInRoot: {
+      expectedInstallerType: InstallerType.Json,
+      inFiles: ["giweights.json"].map(path.normalize),
+      outInstructions: [
+        {
+          type: "copy",
+          source: path.normalize("giweights.json"),
+          destination: path.normalize("engine/config/giweights.json"),
+        },
+      ],
+    },
+    jsonInRandomFolder: {
+      expectedInstallerType: InstallerType.Json,
+      inFiles: [
+        "fold1/",
+        "fold1/giweights.json",
+        "fold1/bumpersSettings.json",
+      ].map(path.normalize),
+      outInstructions: [
+        {
+          type: "copy",
+          source: path.normalize("fold1/giweights.json"),
+          destination: path.normalize("engine/config/giweights.json"),
+        },
+        {
+          type: "copy",
+          source: path.normalize("fold1/bumpersSettings.json"),
+          destination: path.normalize("r6/config/bumpersSettings.json"),
+        },
+      ],
+    },
+    jsonWithFilesInCorrectFolder: {
+      expectedInstallerType: InstallerType.Json,
+      inFiles: [
+        "engine/",
+        "engine/config/",
+        "engine/config/giweights.json",
+        "r6/",
+        "r6/config",
```

Comment:
Throw the known prefixes in constants (and maybe `pathHierarchyFor` for good measure so the input files are as close to what Vortex gives us as possible)?